java/src/main/java/ai/rapids/cudf/ColumnVector.java

```diff
@@ -141,9 +141,14 @@ public ColumnVector(long nativePointer) {
     assert nativePointer != 0;
     offHeap = new OffHeapState(nativePointer);
     MemoryCleaner.register(this, offHeap);
-    this.type = offHeap.getNativeType();
     this.rows = offHeap.getNativeRowCount();
+    DType nativeType = offHeap.getNativeType();
 
+    if (nativeType == DType.DECIMAL32 || nativeType == DType.DECIMAL64) {
```

Comment:
I suggest adding an `EnumSet` for decimal types and a predicate method like `isDecimal` so it easier and cheaper to check for a decimal type.  This is already done for other sets of `DType` values.